nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractComponentNode.java

```diff
@@ -1084,12 +1092,29 @@ public void onParametersModified(final Map<String, ParameterUpdate> updatedParam
         }
 
         // If this component is affected by the Parameter change, we need to re-validate
+        componentAffected |= isConfigurationParameterModified(updatedParameters);
         if (componentAffected) {
             logger.debug("Configuration of {} changed due to an update to Parameter Context. Resetting validation state", this);
             resetValidationState();
         }
     }
 
+    protected void increaseReferenceCounts(final String parameterName) {
+        parameterReferenceCounts.merge(parameterName, 1, (a, b) -> a == -1 ? null : a + b);
+    }
+
+    protected void decreaseReferenceCounts(final String parameterName) {
+        parameterReferenceCounts.merge(parameterName, -1, (a, b) -> a == 1 ? null : a + b);
+    }
+
+    /**
+     * Verifies whether any referenced configuration parameter has been updated and its effective value has changed.
+     * If yes, the component needs to be re-validated.
+     * @param updatedParameters updated parameters
+     * @return true if the component requires re-validation
```

Comment:
With a method named `isConfigurationParameterModified`, it should not return `true` to indicate that the component requires validation. That's simply a side-effect. It should return `true` if any of the given parameters are used in its configuration.

Reply:
Due to refactor this method has been removed